rllib/core/models/torch/primitives.py

```diff
@@ -56,3 +62,106 @@ def __init__(
 
     def forward(self, x):
         return self.mlp(x)
+
+
+class TorchCNN(nn.Module):
```

Comment:
I think structure wise, TorchCNN primitive should just be a stack of Conv2D layers + activations, but TorchCNNEncoder should do the flattening in the end as well. Right now first the primitive is not usable stand-alone if I don't want to flattening portion. This will also clearly separate responsibilities between primitives and encoders i.e. Encoder can be collection of primitives working together. It's not just simple pass-through to one of the primitives.

Reply:
Yeah my idea was to reuse this Conv2d class we had before to be close to what we had before and after seeing it work, simply inline that code. I followed the second part of your advice about the primitive CNN including only the CNN itself. Let's keep the Conv2d CNN just for now to share more code with the legacy models. I'll add a todo to simply inline that code and remove unused bits once the transition has been made, ok?

Reply:
ok makes sense.